### PR TITLE
feat: approve gas token for custom token registration

### PR DESCRIPTION
**Pull request by brtkx**
Closes FS-315

### CODE REVIEW

File: tests/integration/customerc20.test.ts [hunk 1]
```diff
@@ -219,6 +214,36 @@ const registerCustomToken = async (
...
+  }
+
+  const gEstimator = new L1ToL2MessageGasEstimator(l2Signer.provider!)
+  // increase by 500%
+  const maxFeePerGas = (await gEstimator.estimateMaxFeePerGas()).mul(6)
+  // hardcode gas limit to 60k
+  const estimatedGas = BigNumber.from(60_000).mul(maxFeePerGas)
+
+  if (isL2NetworkWithCustomFeeToken()) {
+    const approvalTx = await ERC20__factory.connect(
```
Comment (spsjvc) [inline tests/integration/customerc20.test.ts:240]: This code should be inside `approveGasToken` and `getApproveGasTokenRequest` of `AdminErc20Bridger`

File: src/lib/assetBridger/erc20Bridger.ts [hunk 2]
```diff
@@ -884,6 +885,31 @@ export class AdminErc20Bridger extends Erc20Bridger {
...
+
+      const gEstimator = new L1ToL2MessageGasEstimator(l2Provider)
+
+      // increase by 500%
+      const maxFeePerGas = (await gEstimator.estimateMaxFeePerGas()).mul(6)
+
+      // hardcode gas limit to 60k
+      const estimatedGas = BigNumber.from(60_000).mul(maxFeePerGas)
+
+      if (allowance.lt(estimatedGas)) {
```
Comment (spsjvc) [inline src/lib/assetBridger/erc20Bridger.ts:906]: you don't neeed to use `L1ToL2MessageGasEstimator` + let's use `percentIncrease` (we can copy it over from `L1ToL2MessageGasEstimator` https://github.com/OffchainLabs/arbitrum-sdk/blob/main/src/lib/message/L1ToL2MessageGasEstimator.ts#L76)

```suggestion
      const maxFeePerGasOnL2 = (await l2Provider.getFeeData()).maxFeePerGas
      const maxFeePerGasOnL2WithBuffer = percentIncrease(maxFeePerGasOnL2, 500)

      // hardcode gas limit to 60k
      const estimatedGasFee = BigNumber.from(60_000).mul(maxFeePerGasOnL2WithBuffer)

      if (allowance.lt(estimatedGasFee)) {
```

File: src/lib/assetBridger/erc20Bridger.ts [hunk 3]
```diff
@@ -854,6 +854,65 @@ interface TokenAndGateway {
  * Admin functionality for the token bridge
  */
 export class AdminErc20Bridger extends Erc20Bridger {
+  private percentIncrease(num: BigNumber, increase: BigNumber): BigNumber {
+    return num.add(num.mul(increase).div(100))
+  }
+
+  public getApproveTokenRequestForCustomTokenRegistration(
```
Comment (TucksonDev) [inline src/lib/assetBridger/erc20Bridger.ts:861]: Why do we need this method as well as `getApproveGasTokenRequestForCustomTokenRegistration`?

File: src/lib/assetBridger/erc20Bridger.ts [hunk 4]
```diff
@@ -854,6 +854,52 @@ interface TokenAndGateway {
  * Admin functionality for the token bridge
  */
 export class AdminErc20Bridger extends Erc20Bridger {
+  private percentIncrease(num: BigNumber, increase: BigNumber): BigNumber {
+    return num.add(num.mul(increase).div(100))
+  }
+
+  public getApproveGasTokenRequestForCustomTokenRegistration(
```
Comment (spsjvc) [inline src/lib/assetBridger/erc20Bridger.ts:861]:
```suggestion
  public getApproveGasTokenForCustomTokenRegistrationRequest(
```

File: tests/integration/customerc20.test.ts [hunk 5]
```diff
@@ -219,6 +210,37 @@ const registerCustomToken = async (
...
+    maxFeePerGasOnL2WithBuffer!
+  )
+
+  if (isL2NetworkWithCustomFeeToken()) {
+    await adminErc20Bridger.approveGasTokenForCustomTokenRegistration({
+      l1Signer,
+      amount: estimatedGasFee,
+      erc20L1Address: l1CustomToken.address,
+    })
+  }
```
Comment (spsjvc) [inline tests/integration/customerc20.test.ts:242]: for test simplicity we can just use max

```suggestion
  if (isL2NetworkWithCustomFeeToken()) {
    await adminErc20Bridger.approveGasTokenForCustomTokenRegistration({
      l1Signer,
      erc20L1Address: l1CustomToken.address,
    })
  }
```

**Review by spsjvc — APPROVED** [review]

**Pull request merged**